tsdb/engine/tsm1/engine.go

```diff
@@ -776,7 +776,9 @@ func (e *Engine) Close() error {
 	e.mu.Lock()
 	defer e.mu.Unlock()
 	e.done = nil // Ensures that the channel will not be closed again.
-	e.fieldset.Close()
+	if err := e.fieldset.Close(); err != nil {
```

Comment:
I like not ignoring the error, but perhaps we should still try to close the rest of the internal structures?

Reply:
If we do make that change, I believe we should make the legacy code that follows also close, and not abort the `WAL.Close()` if the `FileStore.Close()` fails, as well.  They seem independent on first glance. Your thoughts?

Reply:
We could just have this as a string of deferred `errors.Capture` calls. That has all the logic to continue and return the first error.

Reply:
That seems reasonable. I'd be happy with either attempting a close on everything, or just keeping the previous program flow but fixing the error swallowing.

Reply:
We now close on everything and return the first error encountered.

https://github.com/influxdata/influxdb/blob/c218c379f13b8fb55f91af84d2cb048fe6d407cb/tsdb/engine/tsm1/engine.go#L780